src/XML/AbstractSignedXMLElement.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\XML;
+
+use DOMElement;
+use SimpleSAML\XMLSecurity\XML\ds\Signature;
+
+/**
+ * Abstract class to be implemented by all signed classes
+ *
+ * @package simplesamlphp/xml-security
+ */
+abstract class AbstractSignedXMLElement implements SignedElementInterface
+{
+    use SignedElementTrait;
+
+    /**
+     * The signed DOM structure.
+     *
+     * @var \DOMElement
+     */
+    protected DOMElement $structure;
+
+    /**
+     * The unsigned elelement.
+     *
+     * @var \SimpleSAML\XMLSecurity\XML\SignableElementInterface
+     */
+    protected SignableElementInterface $element;
+
+
+    /**
+     * Create/parse an alg:SigningMethod element.
+     *
+     * @param \DOMElement $xml
+     * @param \SimpleSAML\XMLSecurity\XML\SignableElementInterface $elt
+     * @param \SimpleSAML\XMLSecurity\XML\ds\Signature $signature
+     */
+    protected function __construct(DOMElement $xml, SignableElementInterface $elt, Signature $signature)
+    {
+        $this->setStructure($xml);
+        $this->setElement($elt);
+        $this->setSignature($signature);
+    }
+
+
+    /**
+     * Output the class as an XML-formatted string
+     *
+     * @return string
+     */
+    public function __toString(): string
+    {
+        return $this->structure->ownerDocument->saveXML();
+    }
+
+
+    /**
+     * Collect the value of the unsigned element
+     *
+     * @return \SimpleSAML\XMLSecurity\XML\SignableElementInterface
+     */
+    public function getElement(): SignableElementInterface
+    {
+        return $this->element;
+    }
+
+
+    /**
+     * Set the value of the elment-property
+     *
+     * @param \SimpleSAML\XMLSecurity\XML\SignableElementInterface $elt
+     */
+    private function setElement(SignableElementInterface $elt): void
+    {
+        $this->element = $elt;
+    }
+
+
+    /**
+     * Collect the value of the structure-property
+     *
+     * @return \DOMElement
+     */
+    public function getStructure(): DOMElement
```

Comment:
I think the getter here is redundant. `toXML()` does the same, and gives as a uniform interface across the library.